### ISSUE TITLE
GSM backend seems to be not working anymore

### ISSUE THREAD

**Issue by sobrus** [n76/Local-GSM-Backend#57]
I've noticed that LocalGSMBackend is not working for me anymore (currently version 1.3.4 from F-Droid). Database is present and is detected by application, created by application or bash script (both lacells creator and new script by wvengen), from Mozilla daily snapshot or pre-filtered CSV.

I've checked database with sqlite - everything seems to be OK, MCC/MNC codes are correct, cell data seems to be OK.

database path is
/sdcard/Android/data/org.fitchfamily.android.gsmlocation/files/lacells.db
(new one)

Still, it cannot determine any position (unified nlp works - tested with Mozilla backend).

Phone is a x86 Zenfone 5, Android 4.3, rooted, without google services.
I can provide more info (including database) if needed, of course.

Where can I download APK older than 1.3.2 for testing? It used to work flawlessly.

**Comment by n76** [comment]
I just checked F-Droid and was surprised they only kept two older versions. But I'd rather figure out what your issue is and fix it anyway.

For a first cut, disable other UnifiedNlp backends, if any, so that only the local GSM Location Service is active. Then, while running some app the uses location services like SatStat give me a few lines of the results of the following:

adb logcat | grep NlpLocationBackendFuser

**Comment by sobrus** [comment]
It's very strange. Backend was not working for at least two days.
Suddenly, when I was capturing logcat, it started to work.
Logcat was showing:

[logcat.txt](https://github.com/n76/Local-GSM-Backend/files/94626/logcat.txt)

Then, I restarted my phone and it doesn't work again.
No log attached, since the log when it's not working is empty.

I've uninstalled all other backends and restarted - still nothing.

Tested with other applications (Maps.ME, Mapfactor Navigator, Location Provider Debugger) - still nothing.

Location Provider Debugger says Network Adapter is ON, but no location info (N/A).

**Comment by n76** [comment]
Sounds like your phone is like mine: It only reports one cell tower to the backend and it only reports it using the older APIs. I've noticed on my phone that the Mozilla backend never includes the cell tower information in its location request and it appears to be that it only uses the newer APIs which return an empty set on my phone.

From your file size comment, I wonder if Mozilla has stopped including the data from OpenCellID. Last time I regenerated my cell database (a week or two ago) it still seemed to have data from its own collection and from OpenCellID.
